tonic/audio_augmentations.py

```diff
@@ -0,0 +1,402 @@
+        samplerate (float): sample rate of the sample
+        sample_length (int): sample length in seconds
+        min_amp (float): minimum of the amplitude range in volts
+        max_amp (float): maximum of the amplitude range in volts
+        factors (float): range of desired factors for amplitude scaling
+        aug_index (int): index of the chosen factor for pitchshift. It will be randomly chosen from the desired range (if not passed while initilization)
+        caching (bool): if we are caching, the DiskCached dataset will dynamically pass copy index of data item to the transform (to set aug_index). Otherwise the aug_index will be chosen randomly in every call of transform
+
+
+    Args:
+        data (np.ndarray): input (single-or multi-channel) signal.
+
+    Returns:
+        np.ndarray: scaled version of the signal.
+    """
+
+    samplerate: float
+    sample_length: float
+    min_amp: float = 0.058
+    max_amp: float = 0.159
+    factors: list = field(
+        default_factory=lambda: list(range(int(1000 * 0.058), int(1000 * 0.159), 10))
+    )
+    aug_index: int = 0
+    caching: bool = False
+
+    def __post_init__(self):
+        # converted to mVolt
+        self.factors = list(
+            range(int(1000 * self.min_amp), int(1000 * self.max_amp), 10)
+        )
+
+    def __call__(self, audio: np.ndarray):
+        if not self.caching:
+            self.aug_index = random.choice(range(0, len(self.factors)))
+
+        max_amplitude = self.factors[self.aug_index] / 1000
+        max_data_amplitude = np.max(np.abs(audio.ravel()))
+        if max_data_amplitude == 0:
+            max_data_amplitude = 1.0
+
+        return audio / max_data_amplitude * max_amplitude
+
+
+@dataclass
+class AddWhiteNoise:
+    """Add white noise to the data sample with a known ratio.
+
+    Parameters:
+        samplerate (float): sample rate of the sample
+        factors (float): range of desired ratios for added noise
+        aug_index (int): index of the chosen factor for noise. It will be randomly chosen from the desired range (if not passed while initilization)
+        caching (bool): if we are caching the DiskCached dataset will dynamically pass copy index of data item to the transform (to set aug_index). Otherwise the aug_index will be chosen randomly in every call of transform
+
+    Args:
+        audio (np.ndarray): data sample
+    Returns:
+        np.ndarray: data sample with added noise
+    """
+
+    samplerate: float
+    factors: list = field(default_factory=lambda: [0.1, 0.2, 0.3, 0.4, 0.5])
+    aug_index: int = 0
+    caching: bool = False
+
+    def __call__(self, audio: np.ndarray):
+        if not self.caching:
+            self.aug_index = random.choice(range(0, len(self.factors)))
+
+        noise = np.random.normal(0, audio.std(), audio.size)
+        noise_factor = self.factors[self.aug_index]
+        noisy_audio = audio + noise_factor * noise
+        return noisy_audio
+
+
+@dataclass
+class AddHomeNoise:
```

Comment:
This is very specific, is there a way how you can abstract it to noise from different datasets?

Reply:
@biphasic I have removed all noise related augmentations from this PR.